### ISSUE TITLE
[MAP] ImportError: DLL load failed while importing EmbeddedCHLone

### ISSUE THREAD

**Issue by Wewhole** [pyCGNS/pyCGNS#146]
To assist reproducing bugs, please include the following:
 * Operating System Windows 10 x64
 * Python version 3.12.6
 * Where Python was acquired system Python 
 * pyCGNS version 6.2.2
 * HDF5 version 1.10.4

I am a beginner and I can successfully import CGNS.MAP through h5py. However, when I import CGNS.MAP through EmbeddedCHLone, I encounter the issue of 'DLL load failed'. I have tried other versions of HDF5 and CGNS and have encountered this issue.

There are two files, “EmbeddedCHLone.py” and “EmbeddedCHLone.cp312-win_amd64.pyd”, in the "path\Python312\Lib\site-packages\pyCGNS-6.2.2-py3.12-win-amd64.egg\CGNS\MAP" folder.

`Traceback (most recent call last):
  File "D:\PythonProject_P\pythonProject1\main.py", line 1, in <module>
    import CGNS.MAP
  File "D:\Program Files\Python312\Lib\site-packages\pycgns-6.2.2-py3.12-win-amd64.egg\CGNS\MAP\__init__.py", line 59, in <module>
    from .EmbeddedCHLone import load, save, probe
ImportError: DLL load failed while importing EmbeddedCHLone`

I would greatly appreciate it if you could answer me!

**Comment by MicK7** [comment]
Hello, thanks for the report.
How did you get pyCGNS on your system ? Did you use conda ?

**Comment by Wewhole** [comment]
> Hello, thanks for the report. How did you get pyCGNS on your system ? Did you use conda ?

Hello, thank you for your reply.

I installed it through source code. Install in the Python 3.12.6 environment using the following code:
`python setup.py build
python setup.py install`
During the installation process, it prompts that hdf5 can be successfully found (I have tested that my hdf5 is installed correctly).

I did not use conda.

**Comment by MicK7** [comment]
Unfortunately, the Python 3.12 version is not well supported and tested yet.
It's even "magical" that it managed to build and install the package.

Which `cython` version and `C compiler` is used during the installation process ? Is it MSVC ?

**Comment by Wewhole** [comment]
> Unfortunately, the Python 3.12 version is not well supported and tested yet. It's even "magical" that it managed to build and install the package.
> 
> Which `cython` version and `C compiler` is used during the installation process ? Is it MSVC ?

I used 'cython 3.0.11' and the C compiler is Visual Studio 2022 (MSVC 14.3).

Could you provide me with the reference versions of the dependent components corresponding to pyCGNS 6.2.2 ? The versions of Python and C compiler, as well as the versions of Cython, HDF5, NumPy, Qt, and VTK

**Comment by Wewhole** [comment]
I have tried to match different versions of the dependencies (python, hdf5, etc) without using conda, but there have been errors in the MAP module and the NAV module cannot be installed correctly.

Finally, I installed all dependencies using conda (conda install conda-forge::package_name) and successfully installed pycgns 6.2.2. It seems that most of the functions are normal. The only one is that after opening `cg_look`, the "vtk view" cannot be used and "Error #500" will appear. The error description is as follows:
_No data or bad data for a VTK display
Cannot create VTK view_
